### PR TITLE
Add Issue requirement to pull request flow

### PR DESCRIPTION
**Pull request by dataclouder**
Pull requests from all contributors should be based upon an
Issue that has already been either originated by the project maintainers
or discussed enough to make clear what the PR should do.

Signed-off-by: Giuseppe Maxia

### CODE REVIEW

**Comment by dataclouder** [comment]
@hodgesrm I thought that using "should be based" we are not imposing but recommending.

Could you suggest a better wording?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,6 +1,17 @@
...
 
+## A Pull Request should be associated with an Issue.
+
+We wish to have discussions in Issues. A single issue may be targeted by multiple PRs.
+If you're offering a new feature or fixing anything, we'd like to know beforehand in Issues,
+and potentially we'll be able to point development in a particular direction.
+
+Related issue: https://github.com/vmware/go-vcloud-director/issues/0123456789
+
+If the PR is a really trivial change (such as a spelling correction) then go ahead and
```
Comment (hodgesrm) [inline .github/PULL_REQUEST_TEMPLATE.md]: Suggested change:  We accept PRs without associated issues provided the change is sufficiently evident from the commit message.  If you have typos or simple bug fixes go for it.

**Comment by dataclouder** [comment]
@hodgesrm I have changed the wording as suggested.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,6 +1,18 @@
 To help us process your pull request efficiently, please replace the 
 comments shown below. 
 
+## A Pull Request should be associated with an Issue.
+
+We wish to have discussions in Issues. A single issue may be targeted by multiple PRs.
+If you're offering a new feature or fixing anything, we'd like to know beforehand in Issues,
+and potentially we'll be able to point development in a particular direction.
+
+Related issue: https://github.com/vmware/go-vcloud-director/issues/0123456789
```
Comment (lvirbalas) [inline .github/PULL_REQUEST_TEMPLATE.md]: The URL doesn't open.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -25,7 +25,9 @@ contributors.  Typical contribution flow steps are:
 
 - Fork the go-vcloud-director repo into a new repo on GitHub
 - Clone the forked repo locally and set the original go-vcloud-director repo as the upstream repo
-- Make changes in a topic branch and commit
+- Open an Issue in go-vcloud-director describing what you propose to do (unless the change is so trivial that an issue is not needed.)
```
Comment (lvirbalas) [inline CONTRIBUTING.md]: Remove the dot before the parentheses.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -25,7 +25,9 @@ contributors.  Typical contribution flow steps are:
...
+- Wait for discussion and possible direction hints in the issue thread.
```
Comment (lvirbalas) [inline CONTRIBUTING.md]: Remove the dot at the end of the sentence to be consistent with the rest of the paragraph.

**Comment by lvirbalas** [comment]
We discussed the changes with Giuseppe. LGTM!

**Pull request merged**